### PR TITLE
refactor: use Node 12 receiveMessageOnPort with Atomics to synchroniz…

### PR DESCRIPTION
**Pull request by alan-agius4**

### CODE REVIEW

File: src/lib/styles/stylesheet-processor.ts [hunk 1]
```diff
@@ -42,10 +41,25 @@ export class StylesheetProcessor {
...
 
-    return css;
+      Atomics.wait(signal, 0, 0);
+
+      const { css, warnings } = receiveMessageOnPort(ioChannel.port2).message;
+
+      warnings.forEach(msg => log.warn(msg));
+      return css;
+    } catch (error) {
+      throw error;
```
Comment (JoostK) [inline src/lib/styles/stylesheet-processor.ts:59]: Such catch, much wow?

File: src/lib/styles/stylesheet-processor.ts [hunk 2]
```diff
@@ -42,10 +41,25 @@ export class StylesheetProcessor {
       browserslistData: this.browserslistData,
     };
 
-    const { css, warnings } = this.styleSheetProcessorWorker(workerOptions);
+    const ioChannel = new MessageChannel();
 
-    warnings.forEach(msg => log.warn(msg));
+    try {
+    const signal = new Int32Array(new SharedArrayBuffer(4));
```
Comment (JoostK) [inline src/lib/styles/stylesheet-processor.ts:47]: Indentation seems off here

**Review by JoostK — COMMENTED** [review]
I don't have any knowledge in this area so can't really review this, but just two nits.

File: src/lib/styles/stylesheet-processor.ts (hunk 1 above)
Comment (alan-agius4) [reply src/lib/styles/stylesheet-processor.ts:59]: 😁

File: src/lib/styles/stylesheet-processor.ts (hunk 2 above)
Comment (alan-agius4) [reply src/lib/styles/stylesheet-processor.ts]: Umm... strange the commit hook should have formatted this.

File: src/lib/styles/stylesheet-processor.ts (hunk 1 above)
Comment (alan-agius4) [reply src/lib/styles/stylesheet-processor.ts]: Forgot that we can use finally without catch

**Pull request merged**